vignettes/detection_efficiency_vignette.Rmd

```diff
@@ -0,0 +1,542 @@
+  map(~ st_buffer(dist = .x$distance, rcv_osc_sf_12)) %>%
+  bind_rows(.id = "distance") %>%
+  st_cast("LINESTRING") %>%
+  dplyr::select(distance, glatos_array, station_no, ins_serial_no, geometry)
+# now view buffer rings
+mapview(rcv_osc_sf) +
+  mapview(buffer_rings)
+```
+
+Now that we have the buffer rings, we are going to pick 3 locations for each distance to potentially deploy range transmitters for 24 hr. We will then create an excel and gpx file of all the locations.
+
+First we transform our buffer rings into spatial points, view them using `mapview()` and filter out the points we want. Then save as an excel and/or gpx file.  
+
+```{r, warning = FALSE}
+#| title: transform, view and filter
+
+buffer_rings_pts <- buffer_rings %>%
+  st_cast("POINT") %>%
+  mutate(
+    id = 1:nrow(.)
+  ) %>%
+  dplyr::select(id, distance:geometry)
+```
+View them in mapview, this part will might take a little bit to select each point/figure out which ones you want. 
+```{r}
+mapview(rcv_osc_sf) +
+  mapview(buffer_rings_pts)
+```
+
+## Select locations for deployment 
+Then filter out the points we want, this will change depending on your study site and what locations you want. We will transform the projection back to WGS 84 as this is likely what your gps or sonar will want. We will also add in a few columns to conform to GLATOS and OTN data standards and rearrange the column order. 
+```{r, results = 'hide'}
+deploy_sites <- buffer_rings_pts %>%
+  st_transform(crs = 4326) %>%
+  filter(id %in% c(
+    4, 49, 155,
+    407, 456, 472,
+    671, 696, 720,
+    835, 876, 920
+  )) %>%
+  rename(
+    receiver_serial_no = ins_serial_no
+  ) %>%
+  mutate(
+    deploy_date_time = NA,
+    deploy_lat = NA,
+    deploy_long = NA,
+    bottom_depth = NA,
+    riser_length = NA,
+    instrument_depth = NA,
+    ins_model_number = NA,
+    ins_serial_no = NA,
+    transmitter = NA,
+    transmitter_model = NA,
+    deployed_by = NA,
+    recovered = NA,
+    recover_date_time = NA,
+    recover_lat = NA,
+    recover_long = NA,
+    data_downloaded = NA,
+    download_date_time = NA,
+    comments = NA,
+    expect_deploy_lat = st_coordinates(.)[, "Y"],
+    expect_deploy_long = st_coordinates(.)[, "X"],
+    comments = NA
+  ) %>%
+  dplyr::select(
+    id:receiver_serial_no,
+    deploy_date_time:expect_deploy_long, geometry
+  )
+```
+
+## Export to excel and gpx formats
+Then we will save as an excel and gpx file. To save as excel we will use [{openxlsx}](https://ycphs.github.io/openxlsx/index.html). You will notice that I don't have it in the load packages area of this vignette. I have not placed it there for the purpose of how R builds vignettes but please add it to your load packages call. You will also need to replace `"YOUR_FILE_PATH"` with your file path for both saving as an excel and/or gpx file. 
+
+```{r, eval = FALSE}
+#| title: save excel and gpx file
+
+# save as excel
+openxlsx::write.xlsx(deploy_sites, "YOUR_FILE_PATH.xlsx")
```

Comment:
So if this chunk is meant to be as an example of what someone could use, the chunk has `echo = TRUE` and `eval = FALSE` so that when it knits the `rmd` it doesn't actual run but when running each chunk I found it also to produce the error. I can remove this all together if you'd like but originally wanted to demonstrate how to export these objects.